ui/src/app/job-details/panels/panels.component.ts

```diff
@@ -156,6 +160,20 @@ export class JobPanelsComponent implements OnInit {
     }, 1500);
   }
 
+  getTopLevelDirectory(): string {
+    if (this.job.extensions && this.job.extensions.tasks) {
+      this.job.extensions.tasks.forEach((task) => {
+        if (this.topLevelExecutionDirectory) return;
+        if (task.callRoot) {
+          const urlParts = task.callRoot.split(this.job.id);
+          this.topLevelExecutionDirectory = urlParts[0] + this.job.id;
```

Comment:
Is the `task.callRoot` guaranteed to contain the `this.job.id`? I only ask because there's an `if` statement above checking if the `task.callRoot` exists, and it seems to be assuming the job.id will therefore also be present in this non-null call root.

Reply:
added check for `job.id` within `callRoot`